lib/chore/job.rb

```diff
@@ -108,11 +115,22 @@ def perform(*args)
 
     # Use the current configured publisher to send this job into a queue.
     def perform_async(*args)
-      self.class.run_hooks_for(:before_publish,*args)
-      @chore_publisher ||= self.class.options[:publisher]
-      @chore_publisher.publish(self.class.prefixed_queue_name,self.class.job_hash(args))
-      self.class.run_hooks_for(:after_publish,*args)
+      push_to_publisher(args)
+    end
+
+    # Use the current configured publisher to send this job into a queue with the given delay period (in seconds). If
+    # the given publisher does not support delayed jobs a warning should be issued and the job will be queued without a
+    # delay.
+    def perform_delayed(delay, *args)
+      push_to_publisher(args, delay: delay)
     end
 
+    private
+    def push_to_publisher(args, options={})
+      self.class.run_hooks_for(:before_publish, *args)
+      @chore_publisher ||= self.class.options[:publisher]
+      @chore_publisher.publish(self.class.prefixed_queue_name, self.class.job_hash(args), options)
```

Comment:
Oh wait, nevermind. Because chore-messaging adds *new* publishers. I'll add a sibling PR over there.

A minor version bump (1.x) will probably be good for this given the API change.